smaily-for-wordpress.php

```diff
@@ -233,19 +244,31 @@ function smaily_nojs_subscribe_callback() {
 
 	$array = array_merge( $array, $form_values );
 	require_once( BP . DS . 'code' . DS . 'Request.php' );
-	$request = new Wp_Sendsmaily_Request( $server, $array );
+	$request = new Wp_Smaily_Request( $server, $array );
 	$result  = $request->exec();
 
 	if ( empty( $result ) ) {
-		echo esc_html__( 'Something went wrong', 'wp_sendsmaily' );
+		$redirect_url .= '?smaily_form_error=' . esc_html__( 'Something went wrong', 'wp_smaily' );
 	} elseif ( (int) $result['code'] !== 101 ) {
-		// Possible errors, for translation.
-		// esc_html__('Posted fields do not contain a valid email address.', 'wp_sendsmaily');.
-		// esc_html__('No autoresponder data set.', 'wp_sendsmaily');.
-		echo esc_html__( $result['message'], 'wp_sendsmaily' );
+		switch ( $result['code'] ) {
+			case 201:
+				$error .= '?smaily_form_error=' . esc_html__( 'Form was not sent using POST method.', 'wp_smaily' );
+				break;
+			case 204:
+				$redirect_url .= '?smaily_form_error=' . esc_html__( 'Input does not contain a recognizable email address.', 'wp_smaily' );
```

Comment:
Use Wordpress' `add_query_arg` method to add query arguments to an URL, and apply URL-encoding to added query arguments when applicable.